File: app/src/main/java/de/cineaste/android/controllFlow/series/WatchlistSeriesTouchHelperCallback.java
package de.cineaste.android.controllFlow.series;

import android.content.Context;
import android.content.res.Resources;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import de.cineaste.android.R;
import de.cineaste.android.adapter.series.SeriesListAdapter;
import de.cineaste.android.controllFlow.BaseSnackBar;

public class WatchlistSeriesTouchHelperCallback extends BaseSeriesTouchHelperCallback {

    private Context context;

    public WatchlistSeriesTouchHelperCallback(Resources resources, LinearLayoutManager linearLayoutManager, RecyclerView recyclerView, SeriesListAdapter seriesListAdapter, Context context) {
        super(resources, linearLayoutManager, recyclerView, seriesListAdapter);
        this.context = context;
    }

    @Override
    protected BaseSnackBar getSnackBar() {
        return new SeriesSnackBarWatchList(linearLayoutManager, recyclerView, seriesListAdapter, context);
    }

    @Override
    protected int getIcon() {
        return R.drawable.ic_add_to_history_white;
    }

    @Override
    protected int getRightSwipeMessage() {
        return R.string.series_seen;
    }
}

File: app/src/main/kotlin/de/cineaste/android/controllFlow/series/WatchlistSeriesTouchHelperCallback.kt
package de.cineaste.android.controllFlow.series

import android.content.Context
import android.content.res.Resources
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import de.cineaste.android.R
import de.cineaste.android.adapter.series.SeriesListAdapter
import de.cineaste.android.controllFlow.BaseSnackBar

class WatchlistSeriesTouchHelperCallback(
    resources: Resources,
    linearLayoutManager: LinearLayoutManager,
    recyclerView: RecyclerView,
    seriesListAdapter: SeriesListAdapter,
    private val context: Context
) : BaseSeriesTouchHelperCallback(resources, linearLayoutManager, recyclerView, seriesListAdapter) {

    override val snackBar: BaseSnackBar
        get() = SeriesSnackBarWatchList(
            linearLayoutManager,
            recyclerView,
            seriesListAdapter,
            context
        )

    override val icon: Int
        get() = R.drawable.ic_add_to_history_white

    override val rightSwipeMessage: Int
        get() = R.string.series_seen
}
